src/Module.Server/Common/Models/CrpgItemValueModel.cs

```diff
@@ -86,20 +88,24 @@ private float CalculateArmorTier(ArmorComponent armorComponent)
 
         return armorComponent.Item.ItemType switch
         {
-            ItemObject.ItemTypeEnum.HorseHarness => 10 * armorPower / bestArmorPower,
-            _ => 10 * armorPower / (bestArmorPower * (float)Math.Pow(armorComponent.Item.Weight + 8, 0.5f)),
+
+            ItemObject.ItemTypeEnum.HorseHarness => 10 * armorPower / bestArmorPower / (float)Math.Pow(1 + heirloomLevel / 10f, 1f),
+            _ => (10 * armorPower) / (bestArmorPower * (float)Math.Pow(armorComponent.Item.Weight + 8, 0.5f)) / (float)Math.Pow(1 + heirloomLevel / 20f, 1f),
         };
     }
 
-    private float CalculateHorseTier(HorseComponent horseComponent)
+    private float CalculateHorseTier(ItemObject item, HorseComponent horseComponent)

```

Comment:
You can the item with `horseComponent.Item` like in the other methods

Reply:
no i can't it throws a null exception

Reply:
Classic